### PR TITLE
Use pundit policies in all necessary controllers

### PR DESCRIPTION
**Pull request by misaka**
This is a security measure to ensure we always scope the DB requests appropriately to the current user.

We also fix the data as many patients were being created without a link to location.

### CODE REVIEW

**Review by benilovj — APPROVED** [review]
Code looks fine but @tvararu should review too

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -9,6 +9,9 @@ class ApplicationController < ActionController::Base
   before_action :set_phase_banner_text
   before_action :authenticate_basic
 
+  after_action :verify_policy_scoped,
+               if: -> { false && Rails.env.development? || Rails.env.test? }
```
Comment (tvararu) [inline app/controllers/application_controller.rb:13]: Is the `false &&` just for testing? 🤔

**Review by tvararu — APPROVED** [review]
1 comment, rest LGTM

File: app/controllers/application_controller.rb (hunk 1 above)
Comment (misaka) [reply app/controllers/application_controller.rb:13]: Aaahh, ya. doh! I was just forcing it to test that path, good spot thanks!

**Comment by misaka** [comment]
I'm just going to ignore the SonarCloud code duplication issue ... I couldn't even make heads or tails of what it was complaining about specifically, never mind how to dismiss it.

**Pull request merged**